### ISSUE TITLE
Remove clever code

### ISSUE THREAD

**Issue by alejandro-colomar** [shadow-maint/shadow#608]
<https://github.com/shadow-maint/shadow/blob/37412f505ed0bc5c9e3f04b2266b40f8f6fc310f/src/groupmod.c#L619>

If I I'm not missing something, this is a buffer overrun in the `snprintf(3)` call.  I have plans to address it, which include defining this function for use in the entire project:

```c
char *
shdw_stpeprintf(char *str, char *past_end, const char *restrict fmt, ...)
{
	int      len;
	char     *p;
	va_list  ap;

	if (str == past_end)
		return past_end;

	if (str == NULL)
		return NULL;

	va_start(ap, fmt);
	len = vsnprintf(str, past_end - str, fmt, ap);
	va_end(ap);

	if (len == -1)
		return NULL;

	if (len >= past_end - str)
		return past_end;

	return str + len;
}
```

This is a wrapper for snprintf(3) which allows easy chaining, and detecting truncation.  Its design is very similar to stpecpy(3), which I designed here: <https://software.codidact.com/posts/285946/287522#answer-287522>

**Comment by alejandro-colomar** [comment]
Ahh, there's no overrun; there was a clever subtraction.  Anyway, I still want to remove the cleverness, to avoid accidents.

**Comment by alejandro-colomar** [comment]
Fixed by #636
